### PR TITLE
move getRequiredRole from Component to ComponentController

### PR DESCRIPTION
**Pull request by matthiaszimmermann**

### CODE REVIEW

File: contracts/modules/ComponentController.sol [hunk 1]
```diff
@@ -196,13 +194,20 @@ contract ComponentController is
...
     }
 
     function getComponentState(uint256 id) public view returns (IComponent.ComponentState componentState) {
         return _componentState[id];
     }
 
+    function getRequiredRole(IComponent.ComponentType componentType) external returns (bytes32) {
+        if (componentType == IComponent.ComponentType.Product) { return _access.productOwnerRole(); }
+        if (componentType == IComponent.ComponentType.Oracle) { return _access.oracleProviderRole(); }
+        if (componentType == IComponent.ComponentType.Riskpool) { return _access.riskpoolKeeperRole(); }
```
Comment (doerfli) [inline contracts/modules/ComponentController.sol:208]: change to if/if else/else and throw exception if other component type (you never know).

File: contracts/modules/LicenseController.sol [hunk 2]
```diff
@@ -20,27 +20,19 @@ contract LicenseController is
         _component = ComponentController(_getContractAddress("Component"));
     }
 
+    // ensures that calling component (productAddress) is a product
```
Comment (doerfli) [inline contracts/modules/LicenseController.sol:23]: this description does not match with the code inside the method.

File: contracts/services/OracleService.sol [hunk 3]
```diff
@@ -18,6 +18,7 @@ contract OracleService is
     }
 
     function respond(uint256 _requestId, bytes calldata _data) external override {
+        // function belw enforces msg.sender to be a registered oracle
```
Comment (doerfli) [inline contracts/services/OracleService.sol:21]:
```
        // function below enforces msg.sender to be a registered oracle
```

**Review by doerfli — CHANGES_REQUESTED** [review]

File: contracts/modules/LicenseController.sol (hunk 2 above)
Comment (matthiaszimmermann) [reply contracts/modules/LicenseController.sol:23]: inside the function _getProduct() is called. if productAddress does not belong to a registered product the function reverts. where do you see a mismatch?

**Pull request merged**